### PR TITLE
Milestone-3: Set up associations

### PR DESCRIPTION
**Pull request by MNisarAli**
## 🎓 Add Classroom, Student, Book, and Rental classes and their relationships 📚

This pull request adds the following changes to the project:

### 🏫 Classroom class:

- `@label` instance variable, with setter and getter methods.
- `has-many` relationship with `Student` class.
- The ability to add students to the classroom and set their classroom.

### 👩‍🎓 Student class:

- `@name` instance variable, with setter and getter methods.
- `belongs-to` relationship with `Classroom` class.

### 📚 Book class:

- `@title` and `@author` instance variables, with setter and getter methods.

### 📅 Rental class:

- `@date` instance variable, with setter and getter methods.
- `belongs-to` relationship with `Book` and `Person` classes.
- `has-many` relationship with `Person` and `Book` classes.
- Modified constructor to set `Book` and `Person` in it.

#### 🙌 Thank you for your consideration!

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  attr_accessor :label, :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
+    student.classroom = self
```
Comment (TadesseDev) [inline classroom.rb]: - [Optional] Dear @MNisarAli, I like your code. I would suggest using some conditionals to check if the student was already inside the `@sudents` array. This is to remove some duplications.
> something like 
```ruby 
@students << student unless @students.include?(self)
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (TadesseDev) [inline book.rb]: - [ ] Dear @MNisarAli, I notice you have  `book.rentals << self` while you initialize rentals, But it's mandatory to add a method to add new rentals to the book. Please consider adding a `add_rental(person, date)` method to add a rental to the current book. [More here about both-way relationship](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

File: person.rb [hunk 3]
```diff
@@ -1,7 +1,8 @@
 require_relative 'nameable'
+require_relative 'rental'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
   attr_accessor :name, :age
```
Comment (TadesseDev) [inline person.rb:7]: - [ ] Dear @MNisarAli, I notice you have  `person.rentals << self` while you initialize rentals, But it's mandatory to add a method to add new rentals to the person. Please consider adding a `add_rental(book, date)` method to add a rental to the current person. [More here about both-way relationship](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @MNisarAli 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [ ]  Correct GitFlow  ✅
 - [ ]  Good commit history  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (MNisarAli) [reply classroom.rb]: Hi @TadesseDev 👋, I am grateful for the valuable suggestion you provided. I have successfully incorporated your suggestion by including a condition to verify if the student already exists in the `@students` array.

File: book.rb (hunk 2 above)
Comment (MNisarAli) [reply book.rb]: Hi @TadesseDev 👋, I am grateful to you for highlighting the need to incorporate the compulsory `add_rental` method. Consequently, I have included the `add_rental(person, date)` method, which will help in adding rentals for the current book.

File: classroom.rb (hunk 1 above)
Comment (TadesseDev) [reply classroom.rb]: Good job @MNisarAli 👌

File: book.rb (hunk 2 above)
Comment (TadesseDev) [reply book.rb]: Good job @MNisarAli 👌

File: person.rb (hunk 3 above)
Comment (MNisarAli) [reply person.rb:7]: Hi @TadesseDev 👋, Again I am grateful to you for highlighting the need to incorporate the compulsory `add_rental` method. Consequently, I have included the `add_rental(book, date)` method, which allows for the seamless addition of rentals for the current person being served.

**Review by AnselemOdims — APPROVED** [review]
### Hi  @MNisarAli 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+require_relative 'rental'
+
+class Book
```
Comment (MNisarAli) [inline book.rb:3]: - [ ] **Use proper naming conventions**: Some variable and function names could be more descriptive to enhance code readability.  For instance, I suggest renaming the `editToDo` function to `handleEditTodo` to provide a clearer indication of its purpose and functionality.

**Pull request merged**